File: MorWalPizVideo.Models/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Runtime.Serialization;

namespace MorWalPizVideo.Server.Models
{    
    [BsonIgnoreExtraElements]
    [DataContract]
    public record Product(
        [property: DataMember][property: BsonElement("title")] string Title,
        [property: DataMember][property: BsonElement("description")] string Description,
        [property: DataMember][property: BsonElement("url")] string Url,
        /// <summary> Categories associated with the product
        /// refers to ProductCategory documents
        /// </summary>
         [property:DataMember]
        [property: BsonElement("categories")]
        CategoryRef[] Categories ) : BaseEntity
{
}

[BsonIgnoreExtraElements]
[DataContract]
public record ProductCategory(
   [property: DataMember][property: BsonElement("title")] string Title,
   [property: DataMember][property: BsonElement("description")] string Description) : BaseEntity
{
}}

File: BackOfficeSPA/back-office-spa/src/models/product.ts
import { CategoryRef } from './video/types';

export interface Product {
  id: string;
  title: string;
  description: string;
  url: string;
  categories: CategoryRef[];
  creationDateTime: string;
}

export interface CreateProductDTO {
  title: string;
  description: string;
  url: string;
  categoryIds: string[];
}

export interface UpdateProductDTO {
  title: string;
  description: string;
  url: string;
  categoryIds: string[];
}
